### PR TITLE
add greatsql addon

### PR DESCRIPTION
**Pull request by ltaodream**
add greatsql addon.
GreatSQL is a high performance open source relational database management system  that can be used on common hardware for financial-grade application scenarios.

### CODE REVIEW

**Comment by shanshanying** [comment]
> add greatsql addon.
> 
> GreatSQL is a high performance open source relational database management system  that can be used on common hardware for financial-grade application scenarios.

Hi @ltaodream 
Is greatsql a variant of MySQL? 
Could you give more description of greatsql and show some instructions to run it along with this PR? (Such as how to install the helm chart how to create a greatsql cluster and how to check it is running )

**Comment by ltaodream** [comment]
> > add greatsql addon.
> > GreatSQL is a high performance open source relational database management system  that can be used on common hardware for financial-grade application scenarios.
> 
> Hi @ltaodream Is greatsql a variant of MySQL? Could you give more description of greatsql and show some instructions to run it along with this PR? (Such as how to install the helm chart how to create a greatsql cluster and how to check it is running )

Yes, it can be seen as  a variant of MySQL.
GreatSQL is a branch of Percona Server, which can be seen as a branch of MySQL. GreatSQL further improves the performance and reliability of MySQL Group Replication (MGR), as well as numerous bug fixes. In addition, InnoDB transaction locks have been optimized and InnoDB parallel query features have been added. GreatSQL can be used as an optional alternative to MySQL or Percona Server for online production environments.

This is its github repository address：https://github.com/GreatSQL/GreatSQL


How to start GreatSQL :
first of all you should git clone kubeblocks-addons and cd to file `addons`

1. helm install greatsql with file greatsql
   `helm install greatsql ./greatsql`
2. helm install greatsql-cluster with file greatsql-cluster
   `helm install greatsql-cluster ./greatsql-cluster`

**Comment by ltaodream** [comment]
> Great job! Could you provide some screenshots of using this addon to run GreatSQL?

image-20240407211616133
image-20240407211635125

So it's ready to use

**Comment by ltaodream** [comment]
> Great job! Could you provide some screenshots of using this addon to run GreatSQL?

greatsql is installed by executing this command

```
kbcli cluster create test --cluster-definition greatsql
```

image

File: addons/greatsql-cluster/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+annotations:
```
Comment (shanshanying) [inline addons/greatsql-cluster/Chart.yaml:1]: pls add more annotations to speicify its dependenc on KB version, such as 
```yaml
  addon.kubeblocks.io/kubeblocks-version: ">=0.8.0"
  addon.kubeblocks.io/model: "RDBMS"
  addon.kubeblocks.io/provider: "community"
```

Comment (ltaodream) [reply addons/greatsql-cluster/Chart.yaml:1]: The annotations of  its dependencies on the KB version are in greatsql's chart.yaml file, not in greatsql-cluster's chart.yaml file. All the other addons in Kubeblock-addons are like this.

File: addons/greatsql/values.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      name: mysqld-exporter
+      tag: v0.14.0
+
+## MySQL Authentication parameters
+auth:
+  ## MYSQL_ROOT_HOST
+  rootHost: "%"
+  ## @param auth.createDatabase Whether to create the .Values.auth.database or not
+  ##
+  createDatabase: true
```
Comment (shanshanying) [inline addons/greatsql/values.yaml:19]: seems values from line 19 to line 32 are not used?

File: addons/greatsql-cluster/values.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+mode: standalone
+
+## @param replicas specify cluster replicas
+##
+replicas: 1
+
+## @param cpu
+##
+cpu: 0.5
```
Comment (shanshanying) [inline addons/greatsql-cluster/values.yaml:11]: please adjust the cpu and memory values properly to make sure the cluster can run under such resource settings.

**Comment by shanshanying** [comment]
Hi @ltaodream 
So, it is a Stand-alone mode Cluster, right? Which means only ONE instance works (not matter how many pods we create, cause I did notice any data sync procedure here)

**Comment by ltaodream** [comment]
> Hi @ltaodream So, it is a Stand-alone mode Cluster, right? Which means only ONE instance works (not matter how many pods we create, cause I did notice any data sync procedure here)

Yes, it now only supports standalone mode like mariadb-cluster in Kubeblock-addons. I tried other databases with complex cluster schema architecture a few days ago, but did not complete its addon because I was not familiar enough with kubeblocks.

So I started with greatsql, a branch of mysql like mariadb and halo. This helped me understand the underlying concept of kubeblocks, the workflow, and how it abstracts the database cluster architecture. 

The greatsql-cluster addon will also support replication mode with two replicas, just like the halo-cluster, or even more. I am familiar with the underlying concept of kubeblocks and will try out addons for other databases with more complex clustering architectures and try to contribute to the kubeblocks master project.

File: addons/greatsql/values.yaml (hunk 2 above)
Comment (ltaodream) [reply addons/greatsql/values.yaml]: Yes, it's for the next version, and I've removed it

File: addons/greatsql-cluster/values.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
+version: greatsql
```
[the remark this answers is not in the document]
Comment (ltaodream) [reply addons/greatsql-cluster/values.yaml]: Here I have changed

**Review by shanshanying — APPROVED** [review]

**Comment by shanshanying** [comment]
image
fix helm lint err pls

**Review by free6om — APPROVED** [review]

**Comment by ltaodream** [comment]
> image fix helm lint err pls

I have fixed！

image

**Pull request merged**